### PR TITLE
fix(dbless): drain error table when flattening errors

### PR DESCRIPTION
**Pull request by flrgh**
This only affect `POST /config?flatten_errors=1`.

When nested errors are flattened for the response body, they should be removed from the original error table so that they are not included in the final `fields` error object. Errors that are not flattened should remain.

It could be argued that this is a backwards-incompatible change, but:

1. This is opt-in only via the `flatten_errors` query param.
2. Because `flatten_errors` is a new, yet unreleased feature, there is no defined behavior to be broken.
3. It's arguably a behavioral fix since we are effectively de-duplicating the errors in the response.
4. This makes the response object much, much cleaner.
5. The overall "shape" of the `fields` object has not changed.

### CODE REVIEW

**Comment by flrgh** [comment]
@RobSerafini This is a fix, but it's not really a blocker for anyone, so feel free to remove from the 3.2 milestone if we're too strapped for time.

**Review by rainest — APPROVED** [review]
No concern from our side. We aren't using `fields` at all (with or without the availability of flattened errors--we do use `messages` for logging though), and I agree that if you've opted into flattened that you shouldn't need the old format also.

**Comment by RobSerafini** [comment]
@tyler-ball - I leave it to you to decide the milestone question.

**Comment by flrgh** [comment]
This will need to wait; removed from milestone.

**Comment by flrgh** [comment]
Converting this back to a draft because I will need to make a changelog edit or two before it can be merged.

**Comment by flrgh** [comment]
Changelog entry was added, and I've just rebased on master. This is good to go once CI is :green_circle:.

**Comment by hbagdi** [comment]
Does this need an explicit cherry pick from CE to EE?

**Comment by hbagdi** [comment]
@flrgh @rainest  Please cherry-pick this PR to EE.

**Pull request merged**
